paperqa/llms.py

```diff
@@ -909,6 +910,94 @@ async def similarity_search(
         )
 
 
+class QdrantVectorStore(VectorStore):
+    client: Any = Field(
+        default=None, description="Instance of qdrant_client.QdrantClient"
+    )
+    collection_name: str = Field(default_factory=lambda: f"paper-qa-{uuid.uuid4().hex}")
+    vector_name: str | None = Field(default=None)
+
+    @model_validator(mode="after")
+    def validate_client(self):
+        from qdrant_client import QdrantClient
+
+        if self.client and not isinstance(self.client, QdrantClient):
+            raise TypeError(
+                f"'client' should be an instance of qdrant_client.QdrantClient. Got {type(self.client)}"
+            )
+
+        if not self.client:
+            self.client = QdrantClient()
+
+        return self
+
+    def clear(self) -> None:
+        super().clear()
+        from qdrant_client import models
```

Comment:
Can we just do this import at the module-level, within a `try`-`except`:

```python
try:
    from qdrant_client import QdrantClient, models
except importError:
    QdrantClient = models = None
```

Then later in `QdrantVectorStore`'s `model_validator` we do something like:

```python
if QdrantClient is None:
    raise importError(...)  # With nice message mentioning the extra
```